yb-voyager/cmd/analyzeSchema.go

```diff
@@ -402,6 +417,148 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
 	}
 }
 
+var unsupportedIndexDatatypes = []string{
+	"citext",
+	"tsvector",
+	"tsquery",
+	"jsonb",
+	"inet",
+	// array as well but no need to add it in the list as fetching this type is a different way TODO: handle better with specific types
+}
+
+func prepareTableToColumnTypeMap(createTableNode *pg_query.Node_CreateStmt) {
+	schemaName := createTableNode.CreateStmt.Relation.Schemaname
+	tableName := createTableNode.CreateStmt.Relation.Relname
+	columns := createTableNode.CreateStmt.TableElts
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	for _, column := range columns {
+		/*
+			e.g. 1. CREATE TABLE public.citext_type (
+					id integer,
+					lists_of_data text[],
+					data public.citext
+				);
+				stmt:{create_stmt:{relation:{schemaname:"public"  relname:"citext_type"  inh:true  relpersistence:"p"  location:258}  table_elts:{column_def:{colname:"id"
+				type_name:{names:{string:{sval:"pg_catalog"}}  names:{string:{sval:"int4"}}  typemod:-1  location:287}  is_local:true  location:284}}  table_elts:
+				{column_def:{colname:"lists_of_data"  type_name:{names:{string:{sval:"text"}}  typemod:-1  array_bounds:{integer:{ival:-1}}  location:315}  is_local:true
+				location:301}}  table_elts:{column_def:{colname:"data"  type_name:{names:{string:{sval:"public"}}  names:{string:{sval:"citext"}}  typemod:-1  location:333}
+				is_local:true  location:328}}  oncommit:ONCOMMIT_NOOP}}  stmt_location:244  stmt_len:108
+
+				2. CREATE TABLE public.ts_query_table (
+					id int generated by default as identity,
+					query tsquery
+				  );
+				stmt:{create_stmt:{relation:{schemaname:"public"  relname:"ts_query_table"  inh:true  relpersistence:"p"  location:211}  table_elts:{column_def:{colname:"id"
+				type_name:{names:{string:{sval:"pg_catalog"}}  names:{string:{sval:"int4"}}  typemod:-1  location:242}  is_local:true  constraints:{constraint:{contype:CONSTR_IDENTITY
+				location:246  generated_when:"d"}}  location:239}}  table_elts:{column_def:{colname:"query"  type_name:{names:{string:{sval:"tsquery"}}
+				typemod:-1  location:290}  is_local:true  location:284}}  oncommit:ONCOMMIT_NOOP}}  stmt_location:196  stmt_len:110
+
+		*/
+		if column.GetColumnDef() != nil {
+			typeNameOnIdx0 := ""
+			typeNameOnIdx1 := ""
+			typeNames := column.GetColumnDef().GetTypeName().GetNames()
+			if len(typeNames) >= 1 { // Names list will have all the parts of qualified type name
```

Comment:
Hmm, I think that should work I earlier thought that its better to just check 0th/1st index but I think its fine to check the last index of the typenames array as it can maximum has length of 2 only in all cases (qualified / unqualifed or native / non-native )
Thanks